### PR TITLE
feat: add ability to use own icon component

### LINKED ISSUE #926: use MaterialCommunityIcons instead of MaterialIcons
## Describe the feature
Use MaterialCommunityIcons instead of MaterialIcons by default in all the components that take the `icon` prop.

## Motivation
I found out MaterialIcons lacks a lot of icons like `logout` of `thumb-up-outline`. But MaterialCommunityIcons looks like MaterialIcons and also have lots of other useful icons. Is there a reason you chose MaterialIcons? at least we can let the devs to select their preferences.

### PR DESCRIPTION
**Pull request by Trancever**
### Motivation

Fix #926 

### Test plan

Run example app on web and native.

### CODE REVIEW

File: src/core/settings.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    style,
+    pointerEvents,
+    accessibilityProps,
+  }: {
+    name: string;
+    color: string;
+    size: number;
+    style: StyleProp<ViewStyle>;
+    pointerEvents: 'none';
+    accessibilityProps: any;
```
Comment (Trancever) [inline src/core/settings.tsx]: @satya164 Is there any way to type this correctly? There is a difference between web and native.

File: src/core/settings.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    color,
+    size,
+    style,
+    pointerEvents,
+    accessibilityProps,
+  }: {
+    name: string;
+    color: string;
+    size: number;
+    style: StyleProp<ViewStyle>;
```
Comment (Trancever) [inline src/core/settings.tsx]: With this approach style will already have `scaleX` set based on direction. Do we want to expose direction prop and let user handle it on his own?

File: src/components/Icon.tsx [hunk 3]
```diff
@@ -7,27 +7,34 @@ import {
...
 } from 'react-native';
+import { Consumer as SettingsConsumer } from '../core/settings';
 
-let MaterialIcons: any;
+let MaterialCommunityIcons: any;
 
 try {
   // Optionally require vector-icons
-  MaterialIcons = require('react-native-vector-icons/MaterialIcons').default;
+  MaterialCommunityIcons = require('react-native-vector-icons/MaterialCommunityIcons')
```
Comment (Trancever) [inline src/components/Icon.tsx]: Maybe we should completely drop support for default Icon set?

**Comment by Trancever** [comment]
@satya164 Can you review this? This is the last thing we need to release 3.0 version.

**Comment by cassioseffrin** [comment]
@Trancever How can I test it en alpha.3 ? How can I set the MaterialIcons instead of
MaterialCommunityIcons ?

That feature will be very appreciated. Mainly because in MaterialCommunityIcons the search icon is not nice, there's not good options for it. And pass the icon as argument is too much verbose.

**Comment by Trancever** [comment]
@cassioseffrin Check this https://github.com/callstack/react-native-paper/wiki/Migration-guide-from-2.0-to-3.0

**Comment by cassioseffrin** [comment]
@Trancever Thks,  I'll checkout!

**Pull request merged**